dask/dataframe/io/parquet/arrow.py

```diff
@@ -938,8 +966,9 @@ def _gather_metadata(
                 ds = pa_ds.parquet_dataset(
                     meta_path,
                     filesystem=fs,
-                    partitioning=dataset_kwargs.get(
-                        "partitioning", default_partitioning
+                    partitioning=partitioning["obj"].discover(
+                        *partitioning.get("args", []),
+                        **partitioning.get("kwargs", {}),
```

Comment:
It looks like we may have dropped using `max_partition_dictionary_size=-1` as a default keyword argument here and in other places we call `discover`

Reply:
Sorry - I think you wrote this while I was working on the [comment below](https://github.com/dask/dask/pull/7066#issuecomment-767156623).  I removed this default value because it is somehow preventing us from using `pickle` on a `ParquetFileFragment` object.  The error does not show up in the following serialization/deserialization, but in the `arrow_table.to_pandas()` call.

As far as I can tell - removing the `max_partition_dictionary_size=-1` is not causing any (correctness) problems, because we are checking the types and manually converting the partition columns to Categorical after the data is read in anyway.  With that said, I'm sure it would be slightly more performant if this option worked.